cannon/cmd/json.go

```diff
@@ -33,10 +34,11 @@ func writeJSON[X any](outputPath string, value X) error {
 		return nil
 	}
 	var out io.Writer
+	var fileUUID = uuid.New()
 	finish := func() error { return nil }
 	if outputPath != "-" {
 		// Write to a tmp file but reserve the file extension if present
-		tmpPath := outputPath + "-tmp" + path.Ext(outputPath)
+		tmpPath := outputPath + "-tmp-" + fileUUID.String() + path.Ext(outputPath)
```

Comment:
I suspect we should do this via `os.CreateTemp` rather than using a UUID.  It will take a bit more fiddling around with this but better to use the builtin functionality to ensure we get a unique file rather than manually reimplementing.

Reply:
I've taken a stab at this in https://github.com/ethereum-optimism/optimism/pull/8193